### ISSUE TITLE
Chef LWRP DSL no longer appreciates "default" resources?

### ISSUE THREAD

**Issue by lonniev** [chocolatey/chocolatey-cookbook#42]
``` bash
WARN: Using an LWRP by its name (Chocolatey) directly is no longer supported in Chef 12 and will be removed.  Use Chef::Resource.resource_for_node(node, name) instead.
```

**Comment by lonniev** [comment]
Has anyone seen an example of how to use `Chef::Resource.resource_for_node(node, name)` correctly?

The cookbooks that I see either don't need to use their `Chef::Resource` class directly or they use the now deprecated way of hardcoding the `Chef::Resource::MyCamelCasedResourceName` class name.

I assume that here `node` should be something like `run_context.node`, also, and `name` should be the `my_camel_cased_resource_name` (clearly underscored rather than CamelCased) symbol as it would appear elsewhere in the DSL.

**Comment by vlnrajesh** [comment]
I too have issue, could someone suggest some examples.

**Comment by epu** [comment]
+1

**Comment by e100** [comment]
```
-  @current_resource = Chef::Resource::Chocolatey.new(@new_resource.name)                         
+  @current_resource = Chef::Resource.resource_for_node(:chocolatey, node).new(@new_resource.name)
```

**Comment by tas50** [comment]
This cookbook no longer ships the package resource since that was included within Chef 12.7 which we released Feb 2016. If you're not yet on Chef 12.7 or later you'll want to upgrade, but you shouldn't see this issue anymore, especially since the 2.0 release of the cookbook removed the package resource entirely.
